Deploy to S3 action failing due to Out of memory error

## Detailed description

After merging the PR's to the master branch, on the action "Deploy website to S3", at the "Build site" step, the action stops and fails after having a `FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory`. A clean text paste of the output was given at the end of this issue, if you can't access the [action direct link](https://github.com/carbon-design-system/design-language-website/actions/runs/13816909703/job/38652385264)

## Steps to reproduce the issue

1. Just try to merge any code on the `master` branch
2. See the "Deploy website to S3" logs

## Additional information

- Screenshots or code
```
Run yarn build:prefix
yarn run v1.22.22
$ gatsby clean && gatsby build --prefix-paths && yarn move-public
info Deleting .cache, public, /home/runner/work/design-language-website/design-language-website/node_modules/.cache/babel-loader, /home/runner/work/design-language-website/design-language-website/node_modules/.cache/terser-webpack-plugin
info Successfully deleted directories
warning Plugin gatsby-plugin-sass-resources is not compatible with your gatsby version 3.14.[6](https://github.com/carbon-design-system/design-language-website/actions/runs/13816909703/job/38652385264#step:7:7) - It requires gatsby@^2.0.0
warning Plugin gatsby-plugin-sass-resources is not compatible with your gatsby version 3.14.6 - It requires gatsby@^2.0.0
success open and validate gatsby-configs, load plugins - 0.96[7](https://github.com/carbon-design-system/design-language-website/actions/runs/13816909703/job/38652385264#step:7:8)s
success onPreInit - 0.003s
success initialize cache - 0.057s
success copy gatsby files - 0.049s
success Compiling Gatsby Functions - 0.160s
Processing SVG metadata
success onPreBootstrap - 0.229s
success createSchemaC***tomization - 0.069s
warning Browserslist: cani***e-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
warning Browserslist: cani***e-lite is outdated. Please run:
  npx update-browserslist-db@latest
  Why you should do it regularly: https://github.com/browserslist/update-db#readme
success Checking for changed pages - 0.000s
success source and transform nodes - 13.404s
success building schema - 0.566s
success createPages - 0.000s
success createPagesStatefully - 3.532s
info Total nodes: 49[8](https://github.com/carbon-design-system/design-language-website/actions/runs/13816909703/job/38652385264#step:7:9)2, SitePage nodes: 55 (***e --verbose for breakdown)
success Checking for changed pages - 0.000s
success update schema - 0.142s
success onPreExtractQueries - 0.000s
success extract queries from components - 0.758s
success write out redirect data - 0.001s
success Build manifest and related icons - 0.431s
success Build manifest and related icons - 0.5[9](https://github.com/carbon-design-system/design-language-website/actions/runs/13816909703/job/38652385264#step:7:10)2s
success onPostBootstrap - 1.317s
info bootstrap finished - 22.591s
success run static queries - 0.087s - [10](https://github.com/carbon-design-system/design-language-website/actions/runs/13816909703/job/38652385264#step:7:11)/10 114.86/s
success run page queries - 0.028s - 55/55 1998.14/s
success write out requires - 0.005s
warning Browserslist: cani***e-lite is outdated. Please run:
  npx update-browserslist-db@latest
  Why you should do it regularly: https://github.com/browserslist/update-db#readme
warning ./node_modules/raw-loader/dist/cjs.js!./src/images/app-icons/dark-theme/RationalDeveloperForI.svg
There are multiple modules with names that only differ in casing.
This can lead to unexpected behavior when compiling on a filesystem with other case-semantic.
Use equal casing. Compare these module identifiers:
* /home/runner/work/design-language-website/design-language-website/node_modules/raw-loader/dist/cjs.js!/home/runner/work/design-language-website/design-language-website/src/images/app-icons/dark-theme/RationalDeveloperForI.svg
    Used by 1 module(s), i. e.
    /home/runner/work/design-language-website/design-language-website/src/images/app-icons|lazy|!!/home/runner/work/design-language-website/design-language-website/node_modules/raw-loader/dist/cjs.js!|/^\.\/.*\/.*\.svg$/|groupOptions: {}|namespace object
* /home/runner/work/design-language-website/design-language-website/node_modules/raw-loader/dist/cjs.js!/home/runner/work/design-language-website/design-language-website/src/images/app-icons/dark-theme/RationalDeveloperforI.svg
    Used by 1 module(s), i. e.
    /home/runner/work/design-language-website/design-language-website/src/images/app-icons|lazy|!!/home/runner/work/design-language-website/design-language-website/node_modules/raw-loader/dist/cjs.js!|/^\.\/.*\/.*\.svg$/|groupOptions: {}|namespace object
warning ./node_modules/raw-loader/dist/cjs.js!./src/images/app-icons/light-theme/RationalDeveloperForI.svg
There are multiple modules with names that only differ in casing.
This can lead to unexpected behavior when compiling on a filesystem with other case-semantic.
Use equal casing. Compare these module identifiers:
* /home/runner/work/design-language-website/design-language-website/node_modules/raw-loader/dist/cjs.js!/home/runner/work/design-language-website/design-language-website/src/images/app-icons/light-theme/RationalDeveloperForI.svg
    Used by 1 module(s), i. e.
    /home/runner/work/design-language-website/design-language-website/src/images/app-icons|lazy|!!/home/runner/work/design-language-website/design-language-website/node_modules/raw-loader/dist/cjs.js!|/^\.\/.*\/.*\.svg$/|groupOptions: {}|namespace object
* /home/runner/work/design-language-website/design-language-website/node_modules/raw-loader/dist/cjs.js!/home/runner/work/design-language-website/design-language-website/src/images/app-icons/light-theme/RationalDeveloperforI.svg
    Used by 1 module(s), i. e.
    /home/runner/work/design-language-website/design-language-website/src/images/app-icons|lazy|!!/home/runner/work/design-language-website/design-language-website/node_modules/raw-loader/dist/cjs.js!|/^\.\/.*\/.*\.svg$/|groupOptions: {}|namespace object
success Building production JavaScript and CSS bundles - 371.245s
success Rewriting compilation hashes - 0.022s
success Writing page-data.json files to public directory - 0.057s - 55/55 958.59/s
<--- Last few GCs --->
[3221:0x27b175f0]   450787 ms: Mark-Compact 4041.7 (4139.8) -> 4029.0 (4143.0) MB, 1872.95 / 0.00 ms  (average mu = 0.093, current mu = 0.029) allocation failure; scavenge might not succeed
[3221:0x27b175f0]   453435 ms: Mark-Compact 4044.8 (4143.0) -> 4032.0 (4146.0) MB, 2593.17 / 0.00 ms  (average mu = 0.052, current mu = 0.021) allocation failure; scavenge might not succeed
<--- JS stacktrace --->
FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory
----- Native stack trace -----
 1: 0xb8d[11](https://github.com/carbon-design-system/design-language-website/actions/runs/13816909703/job/38652385264#step:7:12)b node::OOMErrorHandler(char const*, v8::OOMDetails const&) [/opt/hostedtoolcache/node/20.18.3/x64/bin/node]
 2: 0xf01b70 v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, v8::OOMDetails const&) [/opt/hostedtoolcache/node/20.18.3/x64/bin/node]
 3: 0xf01e57 v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, v8::OOMDetails const&) [/opt/hostedtoolcache/node/20.18.3/x64/bin/node]
 4: 0x1113aa5  [/opt/hostedtoolcache/node/20.18.3/x64/bin/node]
 5: 0x1[12](https://github.com/carbon-design-system/design-language-website/actions/runs/13816909703/job/38652385264#step:7:13)b928 v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [/opt/hostedtoolcache/node/20.18.3/x64/bin/node]
 6: 0x1101a41 v8::internal::HeapAllocator::AllocateRawWithLightRetrySlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [/opt/hostedtoolcache/node/20.18.3/x64/bin/node]
 7: 0x1102bd5 v8::internal::HeapAllocator::AllocateRawWithRetryOrFailSlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [/opt/hostedtoolcache/node/20.18.3/x64/bin/node]
 8: 0x10df2f6 v8::internal::Factory::AllocateRaw(int, v8::internal::AllocationType, v8::internal::AllocationAlignment) [/opt/hostedtoolcache/node/20.18.3/x64/bin/node]
 9: 0x10d0f24 v8::internal::FactoryBase<v8::internal::Factory>::AllocateRawWithImmortalMap(int, v8::internal::AllocationType, v8::internal::Map, v8::internal::AllocationAlignment) [/opt/hostedtoolcache/node/20.18.3/x64/bin/node]
10: 0x10d3706 v8::internal::FactoryBase<v8::internal::Factory>::NewRawOneByteString(int, v8::internal::AllocationType) [/opt/hostedtoolcache/node/20.18.3/x64/bin/node]
11: 0x14249e7 v8::internal::String::SlowFlatten(v8::internal::Isolate*, v8::internal::Handle<v8::internal::ConsString>, v8::internal::AllocationType) [/opt/hostedtoolcache/node/20.18.3/x64/bin/node]
12: 0x14279c0 v8::internal::String::SlowEquals(v8::internal::Isolate*, v8::internal::Handle<v8::internal::String>, v8::internal::Handle<v8::internal::String>) [/opt/hostedtoolcache/node/20.18.3/x64/bin/node]
[13](https://github.com/carbon-design-system/design-language-website/actions/runs/13816909703/job/38652385264#step:7:14): 0x155efc6 v8::internal::Runtime_StringEqual(int, unsigned long*, v8::internal::Isolate*) [/opt/hostedtoolcache/node/20.18.3/x64/bin/node]
[14](https://github.com/carbon-design-system/design-language-website/actions/runs/13816909703/job/38652385264#step:7:15): 0x7f48458d9ef6 
Aborted (core dumped)
error Command failed with exit code 134.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Error: Process completed with exit code 134.
```

--- a/.github/workflows/deploy.yml
+++ b/.github/workflows/deploy.yml
@@ -27,7 +27,7 @@ jobs:
 
       - name: Build site
         env:
-          NODE_OPTIONS: "--max_old_space_size=4096"
+          NODE_OPTIONS: "--max_old_space_size=7000"
         run: yarn build:prefix
 
       - uses: jakejarvis/s3-sync-action@master

